core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala

```diff
@@ -7,18 +7,20 @@ import akka.http.scaladsl.model.headers.OAuth2BearerToken
 import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
 import org.broadinstitute.dsde.rawls.dataaccess.workspacemanager.WorkspaceManagerDAO
 import org.broadinstitute.dsde.rawls.model.ErrorReport
-import bio.terra.workspace.model.{CreatedWorkspace, DataReferenceDescription, WorkspaceDescription}
+import bio.terra.workspace.model.{CreatedWorkspace, DataReferenceDescription, DataReferenceList, WorkspaceDescription}
 import spray.json.{JsObject, JsString}
+import scala.collection.JavaConverters._
 
 import scala.collection.concurrent.TrieMap
 
 class MockWorkspaceManagerDAO extends WorkspaceManagerDAO {
 
-  val references: TrieMap[UUID, DataReferenceDescription] = TrieMap()
+  val references: TrieMap[(UUID, UUID), DataReferenceDescription] = TrieMap()
 
   def mockGetWorkspaceResponse(workspaceId: UUID) = new WorkspaceDescription().id(workspaceId)
   def mockCreateWorkspaceResponse(workspaceId: UUID) = new CreatedWorkspace().id(workspaceId.toString)
-  def mockReferenceResponse(referenceId: UUID) = references.get(referenceId).getOrElse(throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Not found")))
+  def mockReferenceResponse(workspaceId: UUID, referenceId: UUID) = references.get((workspaceId, referenceId)).getOrElse(throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Not found")))
+  def mockEnumerateReferenceResponse(workspaceId: UUID) = new DataReferenceList().resources(references.filterKeys(_._1 == workspaceId).values.toList.asJava)
```

Comment:
ahh, thorny Scala thing here, which is not your fault at all: avoid `filterKeys` and `mapValues`.

Pretty much everywhere else in Scala, filtering/mapping a collection returns a materialized copy of the original with the appropriate transforms applied. `filterKeys` and `mapValues` don't - they return a view, more like Java behavior. This can lead to unexpected consequences and unpredictable bugs.

There are some cases where it's fine, but it's best to just avoid these altogether and never wonder.
https://stackoverflow.com/questions/14882642/scala-why-mapvalues-produces-a-view-and-is-there-any-stable-alternatives
https://github.com/scala/bug/issues/4776

I would rewrite this as:
```
    new DataReferenceList().resources(references.collect {
      case ((wsId, _), refDescription) if wsId == workspaceId => refDescription
    }.toList.asJava)
```

Reply:
Huh, interesting. I'm sure there's a good use for that behavior, but I'll avoid it here.